packages/messaging/index.ts

```diff
@@ -67,13 +60,13 @@ registerMessaging(firebase as _FirebaseNamespace);
  */
 declare module '@firebase/app-types' {
   interface FirebaseNamespace {
-    messaging?: {
-      (app?: FirebaseApp): types.FirebaseMessaging;
-      Messaging: typeof types.FirebaseMessaging;
+    messaging: {
```

Comment:
Soo... @firebase/messaging *is* optional; you can use @firebase/database etc. but not messaging just fine.
firebase the metapackage bundles everything together, cool.
But if you *just* depend on @firebase/messaging, it would monkey-patch @firebase/app-types to add itself optionally? weird